pkg/validation/validation.go

```diff
@@ -0,0 +1,61 @@
+package validation
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// The consts and vars beginning with dns* were taken from: https://github.com/kubernetes/apimachinery/blob/fc49b38c19f02a58ebc476347e622142f19820b9/pkg/util/validation/validation.go
```

Comment:
That function is exported, and we already depended on k8s/apimachinery (https://github.com/dapr/dapr/search?q=apimachinery) so we may as well just call it

Reply:
We might use the same validation logic for other platforms in the future (including self-hosted mode), or other binaries that otherwise would not need those dependencies. Because of the very limited focus of this code I would rather to keep it separate.